packages/algob/src/builtin-tasks/test.ts

```diff
@@ -1,33 +1,24 @@
-import glob from "glob";
-import path from "path";
+import Mocha from "mocha";
 
-import { internalTask } from "../internal/core/config/config-env";
-import { cmpStr } from "../lib/comparators";
-import type { AlgobRuntimeEnv, ResolvedAlgobConfig } from "../types";
-import { TaskArgs } from "./deploy";
-import { TASK_TEST_GET_TEST_FILES } from "./task-names";
+import { task } from "../internal/core/config/config-env";
+import { testsDirectory } from "../lib/script-checkpoints";
+import type { AlgobConfig } from "../types";
+import { loadFilenames } from "./deploy";
+import { TASK_TEST } from "./task-names";
 
-export default function (): void {
-  internalTask(TASK_TEST_GET_TEST_FILES)
-    .addOptionalVariadicPositionalParam(
-      "filenames",
-      "An optional list of files to test",
-      []
-    )
-    .setAction(runTests);
-}
-
-function loadFiles (fileNames: string[], config: ResolvedAlgobConfig): string[] {
-  if (fileNames.length !== 0) {
-    return fileNames;
-  }
-  if (config.paths == null) {
-    throw new Error("unexpected non-project execution");
-  }
-  return glob.sync(path.join(config.paths.tests, "**/*.js")).sort(cmpStr);
+async function runTests (config: AlgobConfig): Promise<void> {
+  const testFiles = loadFilenames(testsDirectory);
+  console.log("Test files:", testFiles);
+  const mocha = new Mocha(config.mocha);
+  // Adding test files to mocha object
+  testFiles.forEach((file) => mocha.addFile(file));
+  const testFailures = await new Promise<number>((resolve) => {
+    mocha.run(resolve);
+  });
+  console.log(testFailures);
```

Comment:
I don't think we need this log. `mocha.run` shows test failures very intuitively (as showed on call).